uprotocol/transport/ulistener.py

```diff
@@ -18,13 +18,25 @@
 
 
 class UListener(ABC):
-    """For any implementation that defines some kind of callback or function that will be called to handle incoming
+    """
+    For any implementation that defines some kind of callback or function that will be called to handle incoming
     messages.
     """
 
     @abstractmethod
-    def on_receive(self, umsg: UMessage) -> None:
-        """Method called to handle/process messages.<br><br>
+    async def on_receive(self, umsg: UMessage) -> None:
```

Comment:
This should be synchronous API no?

Reply:
If we make it synchronous, we can't use await inside it. Since the send API is asynchronous, it must be awaited to function correctly. During testing with SOME/IP, Divya reported that we were not getting the RPC response because transport.send wasn't awaited in the on_receive method.